pkg/libvmi/vmi.go

```diff
@@ -289,6 +290,18 @@ func WithTPM(persistent bool) Option {
 	}
 }
 
+func WithResourceVersion(resourceVersion string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		vmi.ObjectMeta.ResourceVersion = resourceVersion
+	}
+}
+
+func WithUID(uid k8stypes.UID) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		vmi.UID = uid
+	}
+}
+
```

Comment:
Can you keep these private for now? These are unusual functions which users outside of testing will typically not use.